### PR TITLE
[C#] Added vector constructors to allow create them from one scalar

### PR DESCRIPTION
**Pull request by Chaosus**
Allows to create a vector like:

`var v = new Vector2i(64); // 64x64 vector declaration`

I know it's a syntax sugar but I found it's useful for myself.

### CODE REVIEW

File: modules/mono/glue/GodotSharp/GodotSharp/Core/Vector2.cs [hunk 1]
```diff
@@ -581,6 +581,16 @@ public Vector2(real_t x, real_t y)
             this.y = y;
         }
 
+        /// <summary>
+        /// Constructs a new <see cref="Vector2"/> from a scalar assigned to all components.
```
Comment (aaronfranke) [inline modules/mono/glue/GodotSharp/GodotSharp/Core/Vector2.cs:585]: I think we can use more common words here to be more beginner friendly.

```suggestion
        /// Constructs a new <see cref="Vector2"/> from a single value assigned to all components.
```

File: modules/mono/glue/GodotSharp/GodotSharp/Core/Vector2.cs [hunk 2]
```diff
@@ -581,6 +581,16 @@ public Vector2(real_t x, real_t y)
...
+        /// </summary>
+        /// <param name="xy">The vector's X and Y components.</param>
+        public Vector2(real_t xy)
+        {
+            this.x = xy;
```
Comment (aaronfranke) [inline modules/mono/glue/GodotSharp/GodotSharp/Core/Vector2.cs:590]: Unnecessary `this.`

**Review by aaronfranke — CHANGES_REQUESTED** [review]

File: modules/mono/glue/GodotSharp/GodotSharp/Core/Vector2.cs (hunk 1 above)
Comment (Chaosus) [reply modules/mono/glue/GodotSharp/GodotSharp/Core/Vector2.cs]: Fixed

File: modules/mono/glue/GodotSharp/GodotSharp/Core/Vector2.cs (hunk 2 above)
Comment (Chaosus) [reply modules/mono/glue/GodotSharp/GodotSharp/Core/Vector2.cs]: Fixed

File: modules/mono/glue/GodotSharp/GodotSharp/Core/Vector3.cs [hunk 3]
```diff
@@ -610,6 +610,17 @@ public Vector3(real_t x, real_t y, real_t z)
             this.z = z;
         }
 
+        /// <summary>
+        /// Constructs a new <see cref="Vector3"/> from a single value assigned to all components.
+        /// </summary>
+        /// <param name="xy">The vector's X, Y and Z components.</param>
```
Comment (aaronfranke) [inline modules/mono/glue/GodotSharp/GodotSharp/Core/Vector3.cs:616]:
```suggestion
        /// <param name="xyz">The value to assign to all components.</param>
```

**Review by aaronfranke — CHANGES_REQUESTED** [review]

**Comment by aaronfranke** [comment]
Also, this shouldn't be in C# unless it's also in core. Currently `print(Vector3(5))` is not valid in GDScript.

**Comment by Chaosus** [comment]
I know, but I need to found a way to implement it properly since it suddenly has compilation conflicts with other constructors in GDScript...

**Comment by Chaosus** [comment]
Need to think about how to do it for GDScript, currently, I lose interest to do it, maybe later I create another PR with all the requested changes been made.

**Pull request closed without merging**